### PR TITLE
Replace uses of history.imp and history.copy web methods for API create

### LINKED ISSUE #1135: Preserve annotation and tags when copying/importing datasets
Enhancement request:
- When import importing or copying a dataset with tags, retain tags. (Currently are stripped).
- In the "history view" of a shared or published history containing a dataset with tags, show the tags in the expanded dataset.
- Think about ideas for how a dataset tag could be inherited by a history tag list or at least be summarized in the "history view" right panel. (Currently only history tags are shown).

Moved from Trello:
- 6/2013 issue report https://trello.com/c/R1yf4N6M
- 2/2014 issue report https://trello.com/c/Og0lP3J3

### LINKED ISSUE #1291: Unify history copy/import functions
Right now we use three different urls/methods to duplicate a history. I'd like to reduce that to one: the API create method. This would help add things like: adding persistence of tags and annotations when copying histories between users, etc.

Steps:
- [x] create a control to allow the user to copy/import, set the name, choose all datasets or just non-deleted (as the copy page does now)
- [x] replace the history/copy function in the history options menu
- [x] replace the history/copy function in the 'saved histories' grid
- [x] replace the history/copy function in the 'histories shared with me' grid
- [x] replace the history/copy (multiple) in the 'histories shared with me' grid as well
- [x] replace the history/imp function in history/view
- [x] replace the history/imp function in the published history page
- [x] replace the history/imp function in the Page embedded histories

(Default) naming should stay the same (e.g. 'imported :', 'Copy of ', 'shared by ...').
- [x] test with histories containing large numbers of datasets should be done as well

### PR DESCRIPTION
**Pull request by carlfeberhard**
Resolves #1291 

Using grep, I looked for all areas in lib, templates, and client for links to and uses of `controllers/history.copy` and `.imp` and replaced them with the dialog used in the multi-history page. These included:
- history/display.mako (published history display)
- history/view.mako
- history/embed.mako
- both the saved and 'shared with me' history grids
- history/copy.mako and the history options menu

Here's a shorter summary of changes:
- mvc/history/copy-dialog was expanded (and split to still allow the use of the subtly distinct 'Import' language). 
- The grids were changed to use individual mako templates to insert the javascript options dialog and remove the original url behavior. (This is particularly hacky but was the shortest distance between two points I could see at the moment.)
- An effort was made to preserve language, behavior, and especially (default) naming of imports/copies ('imported: ', 'shared by', etc.)
- finally both `imp` and `copy` were removed from the history controller

This should allow **proper copying/persistence of tags & annotations** when copying your own or someone else's (fixes #1135).

This **changes the API** to allow anon users to copy or create new histories (which replace their current).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/histories.py [hunk 1]
```diff
@@ -246,6 +246,7 @@ def published( self, trans, **kwd ):
             rval.append( history_dict )
         return rval
 
+    # TODO: does this need to be anonymous_and_sessionless? Not just expose_api?
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/histories.py:249]: For reference, this was added in https://github.com/galaxyproject/galaxy/pull/614.

**Comment by jmchilton** [comment]
Tested and it seems solid to me :+1:. @carlfeberhard I tried to merge this against dev myself but I can't rebuild the bundles - does `cd client; grunt` not replace the bundles?

**Comment by dannon** [comment]
@jmchilton Have to grunt _and_ webpack right now.  (webpack does the bundling)

**Comment by dannon** [comment]
+1, merged manually to resolve conflicts.

**Pull request merged**